cpp/lib/src/LobidUtil.cc

```diff
@@ -0,0 +1,120 @@
+/*  \brief Utility Functions for Lobid API (HBZ) to get title data, norm data (GND) and organisational data
+ *  \copyright 2018 Universitätsbibliothek Tübingen.  All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "LobidUtil.h"
+#include <unordered_map>
+#include "Downloader.h"
+#include "JSON.h"
+#include "UrlUtil.h"
+#include "util.h"
+
+
+namespace LobidUtil {
+
+
+std::unordered_map<std::string, const std::shared_ptr<const JSON::ObjectNode>> url_to_lookup_result_cache;
+
+
+static std::string BASE_URL_GND = "http://lobid.org/gnd/search?format=json";
+static std::string BASE_URL_ORGANISATIONS = "http://lobid.org/organisations/search?format=json";
+static std::string BASE_URL_RESOURCES = "http://lobid.org/resources/search?format=json";
+
+
+const std::string BuildUrl(const std::string &base_url, const std::unordered_map<std::string, std::string> &params) {
+    std::string url(base_url);
+    unsigned i(0);
+    for (const auto &key_and_value : params) {
+        url += "&";
+        if (i == 0)
+            url += "q=";
+        url += key_and_value.first + UrlUtil::UrlEncode(":" + key_and_value.second);
+        ++i;
+    }
+    return url;
+}
+
+
+const std::shared_ptr<const JSON::ObjectNode> Query(const std::string &url, const bool allow_multiple_results) {
+    LOG_DEBUG(url);
+    const auto url_and_lookup_result(url_to_lookup_result_cache.find(url));
+    if (url_and_lookup_result != url_to_lookup_result_cache.end())
+        return url_and_lookup_result->second;
+
+    Downloader downloader(url);
+    if (downloader.anErrorOccurred())
+        LOG_ERROR(downloader.getLastErrorMessage());
+
+    std::shared_ptr<JSON::JSONNode> root_node(nullptr);
```

Comment:
I don't think that you need to initialise this variable.  Also, you should move it down a line to be as close to its initial use as possible.